src/macaron/util.py

```diff
@@ -254,8 +254,10 @@ def get_if_exists(doc: JsonType, path: list[str | int]) -> JsonType | None:
     """Get a json dict value if it exists."""
     while len(path) > 0:
         this = path.pop(0)
-        if isinstance(doc, (dict, list)) and this in doc:
-            doc = doc[this]  # type: ignore
+        if isinstance(this, str) and isinstance(doc, dict) and this in doc:
+            doc = doc[this]
+        elif isinstance(this, int) and isinstance(doc, list) and this in doc:
```

Comment:
Hmm I think in this branch, we are trying to extract a nth element from `doc` if `doc` is a list. 
I don't think we should run `this in doc` as this means that if `this` is contained in `doc` (a list can contains an int). I guess we should use `this < len(doc)` instead ? :thinking:

Reply:
You are correct, and this is the same logic as previously, which means it was always incorrect for lists. I guess if this function is not currently being used then that wasn't noticed. Regardless, I have fixed it to check if the index is within the range of the list.